x-pack/plugin/search-business-rules/src/test/java/org/elasticsearch/xpack/searchbusinessrules/PinnedQueryBuilderIT.java

```diff
@@ -141,6 +142,28 @@ public void testPinnedPromotions() throws Exception {
 
     }
 
+    public void testIssue51034() throws Exception {
```

Comment:
nit I prefer explicit names that don't require looking up github issues to understand what the test is testing
```suggestion
    /**
     * Test scoring the entire set of documents, which uses a slightly different logic when creating scorers.
     */
    public void testExhaustiveScoring() throws Exception {
```